master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/ha/StateMachine.java

```diff
@@ -222,6 +223,18 @@ protected ResourceResponse runCommand(ResourceProtos.ResourceRequest request, lo
     return null;
   }
 
+  @VisibleForTesting
+  protected org.apache.celeborn.common.protocol.PbMetaRequestResponse runCommand(
+      PbMetaRequest request, long trxLogIndex) {
+    try {
+      return metaHandler.handleWriteRequest(request);
+    } catch (Throwable e) {
+      String errorMessage = "Request " + request + "failed with exception";
+      ExitUtils.terminate(1, errorMessage, e, LOG);
+    }
+    return null;
+  }
+
   /** Query the state machine. The request must be read-only. */
   @Override
   public CompletableFuture<Message> query(Message request) {
```

Comment:
As you can see that the resource protobuf structures are serialized to byteString and sent to the Ratis network, which means that you can deserialize them as transport protobuf structures as long as the protobuf structures are compatible.